app/javascript/controllers/sort_filter_menu_controller.js

```diff
@@ -2,59 +2,87 @@
 
-  applyFilter(filterValue) {
-    const currentUrl = new URL(window.location.href)
-    const [param, value] = filterValue.split('=')
-  
-    this.removeExistingParams(currentUrl)
-    currentUrl.searchParams.set(param, value)
-    window.location.href = currentUrl.toString()
+    this.filterOptionTargets.forEach(option => {
+      const [param, value] = option.value.split('=')
+      if (currentUrl.searchParams.get(param) === value) {
+        option.checked = true
+
+        // only one radio button checked at a time
+        this.filterOptionTargets.forEach(otherOption => {
+          if (otherOption !== option && otherOption.checked) {
+            otherOption.checked = false
+          }
+        })
+      }
+    })
+
+    // preserve submission id in search input field when page loads/refreshes
+    if (this.hasSubmissionIdSearchTarget && currentUrl.searchParams.has('submission_id')) {
+      const submissionId = currentUrl.searchParams.get('submission_id')
+      this.submissionIdSearchTarget.value = submissionId
+      this.searchTerm = submissionId
+    }    
   }
 
-  clearAllFilters() {
-    const currentUrl = new URL(window.location.href)
-    this.removeExistingParams(currentUrl)
-    window.location.href = currentUrl.toString()
+  handleSearchInput(event) {
+    this.searchTerm = event.target.value.trim()
   }
 
-  removeExistingParams(url) {
-    const paramsToRemove = ['status', 'eligible_for_evaluation', 'selected_to_advance', 'sort']
-    paramsToRemove.forEach(key => {
-      url.searchParams.delete(key)
+  handleRadioChange(event) {
+    const clickedRadio = event.target
+
+    this.filterOptionTargets.forEach(radio => {
+      if (radio !== clickedRadio) {
+        radio.checked = false
+      }
     })
   }
 
-  updateCheckmarks() {
-    const currentUrl = new URL(window.location.href)
-    this.menuItemTargets.forEach(item => {
-      const checkmark = item.querySelector('.checkmark')
-      const [param, value] = item.dataset.filterValue.split('=')
-      if (currentUrl.searchParams.get(param) === value) {
-        checkmark.style.display = 'inline'
-      } else {
-        checkmark.style.display = 'none'
-      }
+  applyFilters() {
+    const selectedFilters = this.filterOptionTargets
+      .filter(radio => radio.checked)
+      .map(radio => radio.value)
+
+    let queryParams = selectedFilters
+
+    if (this.searchTerm) {
+      queryParams = queryParams ? 
+        `${queryParams}&submission_id=${this.searchTerm}` :
+        `submission_id=${this.searchTerm}`
+    }
+
+    window.location.href = `${window.location.pathname}?${queryParams}`
+  }
+  
+  clearAllFilters(event) {
+    if (event) {
+      event.preventDefault()
+    }
+    
+    if (this.hasSubmissionIdSearchTarget) {
+      this.submissionIdSearchTarget.value = ''
+      this.searchTerm = ''
+    }
+    
+    this.filterOptionTargets.forEach(radio => {
+      radio.checked = false
     })
+
+    window.location.replace(window.location.pathname)
```

Comment:
this is ok but the window.location update will reset all the search/filter params when the page is reloaded so the extra state clearing isn't really needed. imo the new location should go in the history instead of replacing it.

```suggestion
    window.location.href = window.location.pathname
```